dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -869,6 +878,15 @@ public List<BlockWorkerInfo> getCachedWorkers() throws IOException {
    * @return the info of all block workers
    */
   protected List<BlockWorkerInfo> getAllWorkers() throws IOException {
+    try (ReinitBlockerResource r = blockReinit()) {
```

Comment:
I think I have reasons to believe this `blockReinit()` gives you thread safety against `reinit()`. But my gut feeling is we should:
1. Either remove ConfigHashSync together with `reinit()`
2. Or add a UT on thread safety, in a follow-up PR

Could you add a TODO on this?

Reply:
I will address this as part of subsequent changes. will add TODO to notify

Reply:
[ADDRESS IN NEXT PR]